### DISCUSSION TITLE
Information required

### DISCUSSION THREAD

**Discussion by biplab123321** [aws/aws-connected-device-framework#85] [General]
We have IoT Device Management service, how Is CDF different from it, or do both complement each other?
What's the advantage of using CDF over the service?

**Comment by ayehia** [comment]
I also have the same question and don't know exactly what the CDF brings to the table differently from IoT device management service.

**Comment by aaronatbissell** [comment]
From our experience with CDF, it is a compliment to the IoT Device Management service. It consists as a suite of pre-built [modules](https://github.com/aws/aws-connected-device-framework#architecture) that sit on top of other AWS services (IoT Device Management included). I would recommend taking a look at a few of the modules CDF provides and reading their descriptions to see how they compliment the other AWS Services. Maybe start with [provisioning](https://github.com/aws/aws-connected-device-framework#provisioning) or [asset library](https://github.com/aws/aws-connected-device-framework#asset-library) or [commands](https://github.com/aws/aws-connected-device-framework#commands). Those descriptions pretty clearly lay out how those CDF modules augment or compliment other AWS Services.

Hope this helps!

**Reply by ayehia** [reply]
Thanks for your clarification

I noticed from their documentation that we should use CDF if we are going to implement our facade "consuming applications" that will implement unique business logic and call the CDF modules as microservices to interact with AWS services

<img width="928" alt="image" src="https://user-images.githubusercontent.com/12626365/210788544-5a90e679-4330-46b4-a654-28d8408d8a16.png">

So I think the conclusion is we can directly use the IoT device management services including "Fleet Hub" if we have no unique business logic and use CDF modules if we have unique business logic and will implement our own consuming applications instead of using Fleet Hub managed web apps. 

What do you think?

**Reply by aaronatbissell** [reply]
I don't think that is the intent.

The CDF modules are meant to enhance the functionality of the pure AWS services. If you don't need the enhanced functionality, then you don't need CDF. You can just use the pure AWS services with your application sitting on top.

For instance, we use the Asset Library service because have many different types of assets in our system (devices, firmware, users, etc) that are all connected (a device has firmware and is connected to a user). The Asset Library service enhances IoT Device Management and any replaces any database services you were using to keep track of all that data and gives you and interface to CRUD all the devices, firmware, users, etc.

In our instance, we replaced a couple of dynamo tables that referenced each other with the Asset Library service.
